src/ui/public/parse_query/lib/to_user.js

```diff
@@ -10,6 +10,7 @@ export default function toUser(text) {
   if (text == null) return '';
   if (_.isObject(text)) {
     if (text.query_string) return toUser(text.query_string.query);
+    if (text.match_all && !_.has(text.match_all, 'boost')) return '';
     return angular.toJson(text);
   }
   if (text === '*') {
```

Comment:
If the default is now match_all, is this part necessary? I understand there are saved searches that already have "\*" as the query, but with your changes, if you delete the "\*" then you'll have an empty query (so the placeholder can show) and the default will no longer be "\*". I think it would be more intuitive if we ripped this out with these changes.